src/backend/distributed/utils/node_metadata.c

```diff
@@ -620,27 +684,44 @@ AddNodeMetadata(char *nodeName, int32 nodePort, int32 groupId, char *nodeRack,
 		}
 	}
 
+	/* if nodeRole hasn't been added yet there's a constraint for one-node-per-group */
+	if (nodeRole != InvalidOid)
+	{
+		primaryRole = PrimaryNodeRoleId();
+		if (nodeRole == primaryRole)
```

Comment:
Yeah, what's our convention here? We usually keep it separated but sometimes inline; I've assumed that it's to aid using gdb and tried to break it out whenever a future debugger might want to know the intermediate value, should I be using a different metric?